backend/src/v2/books/repository.ts

```diff
@@ -0,0 +1,227 @@
+import { Like } from "typeorm";
+import { dateAddDays, dateFormat } from "~/kysely/sqlDates";
+
+export const vSearchBookRepo = jipDataSource.getRepository(VSearchBook)
+export const bookRepo = jipDataSource.getRepository(Book);
+export const bookInfoRepo = jipDataSource.getRepository(BookInfo);
+export const vSearchBookByTagRepo = jipDataSource.getRepository(VSearchBookByTag);
+
+export const getBookInfosByTag = async (whereQuery: object, sortQuery: object, page: number, limit: number) => {
+	return await vSearchBookByTagRepo.findAndCount({
+		select: [
+			'id',
+			'title',
+			'author',
+			'isbn',
+			'image',
+			'publishedAt',
+			'createdAt',
+			'updatedAt',
+			'category',
+			'superTagContent',
+			'subTagContent',
+			'lendingCnt'
+		],
+		where: whereQuery,
+		take: limit,
+		skip: page * limit,
+		order: sortQuery
+	});
+}
+
+export const getBookInfosSorted = (limit: number) =>
+	db
+		.selectFrom('book_info')
+		.leftJoin('book', 'book_info.id', 'book.infoId')
+		.leftJoin('category', 'book_info.categoryId', 'category.id')
+		.leftJoin('lending', 'book.id', 'lending.bookId')
+		.select([
+			'book_info.id as id',
+			'book_info.title as title',
+			'book_info.author as author',
+			'book_info.publisher as publisher',
+			'book_info.isbn as isbn',
+			'book_info.image as image',
+			'category.name as category',
+			'book_info.publishedAt as publishedAt',
+			'book_info.createdAt as createdAt',
+			'book_info.updatedAt as updatedAt',
+		])
+		.select(({ eb }) => eb.fn.count('lending.id').as('lendingCnt'))
+		.limit(limit)
+		.groupBy('id')
+
+export const searchBookInfoSpecById = async ( id: number ) =>
+	db
+		.selectFrom('book_info')
+		.select([
+			'id',
+			'title',
+			'author',
+			'publisher',
+			'isbn',
+			'image',
+			'publishedAt',
+			'createdAt',
+			'updatedAt'
+		])
+		.select(({ selectFrom }) => [
+			selectFrom('category')
+			.select('name')
+			.whereRef('category.id', '=', 'book_info.categoryId')
+			.as('category'),
+		])
+		.where('id', '=', id)
+		.executeTakeFirst();
+
+export const searchBooksByInfoId = async ( id: number ) =>
+	db
+		.selectFrom('book')
+		.select([
+			'id',
+			'callSign',
+			'donator',
+			'status'
+		])
+		.where('infoId', '=', id)
+		.execute();
+
+export const getIsLendable = async (id: number) =>
+{
+	const count1 = await db
+					.selectFrom('lending')
+					.where('bookId', '=', id)
+					.where('returnedAt', 'is', null)
+					.select(({ eb }) => eb.fn.countAll().as('count1'))
+					.executeTakeFirst();
+
+	const count2 = await db
+					.selectFrom('book')
+					.where('id', '=', id)
+					.where('status', '=', 0)
+					.select(({ eb }) => eb.fn.countAll().as('count2'))
+					.executeTakeFirst();
+
+	const count3 = await db
+					.selectFrom('reservation')
+					.where('bookId', '=', id)
+					.where('status', '=', 0)
+					.select(({ eb }) => eb.fn.countAll().as('count3'))
+					.executeTakeFirst();
```

Comment:
```suggestion
	const isLended = await db
					.selectFrom('lending')
					.where('bookId', '=', id)
					.where('returnedAt', 'is', null)
					.executeTakeFirst();

	const book = await db
					.selectFrom('book')
					.where('id', '=', id)
					.where('status', '=', 0)
					.executeTakeFirst();

	const isReserved = await db
					.selectFrom('reservation')
					.where('bookId', '=', id)
					.where('status', '=', 0)
					.executeTakeFirst();

	return book && !isLended && isReserved;
```

1. count1, count2보다는 원하는 동작을 반영한 변수명을 사용하면 이해에 도움이 될 것 같아요.
2. 컬럼의 수가 아닌 존재 유무만 확인하는 것이기 때문에, `executeTakeFirst`만 써서 구현하면 코드를 더 간략화할수 있을 것 같아요.

Reply:
확실히 변수명에 원하는 동작을 반영하니 보기 편하군요!!